### PR TITLE
upgrade solidity to version v0.8.11

### PR DESCRIPTION
**Pull request by noobiescoder**

### CODE REVIEW

**Comment by Grimler91** [comment]
Looks like the linker to use might be hardcoded somewhere:

```
2022-02-03T09:47:14.8484596Z [ 60%] Building CXX object tools/CMakeFiles/phaser.dir/yulPhaser/ProgramCache.cpp.o
2022-02-03T09:47:15.5766245Z 130 warnings generated.
2022-02-03T09:47:15.6037506Z [ 61%] Linking CXX executable solidity-upgrade
2022-02-03T09:47:15.6983772Z /usr/bin/ld.gold: fatal error: /home/builder/.termux-build/_cache/android-r23b-api-24-v6/sysroot/usr/lib/aarch64-linux-android/24/crtbegin_dynamic.o: unsupported ELF machine number 183
2022-02-03T09:47:15.7220903Z clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
2022-02-03T09:47:15.7244812Z make[2]: *** [tools/CMakeFiles/solidity-upgrade.dir/build.make:189: tools/solidity-upgrade] Error 1
2022-02-03T09:47:15.7252724Z make[1]: *** [CMakeFiles/Makefile2:696: tools/CMakeFiles/solidity-upgrade.dir/all] Error 2
```

**Comment by xtkoba** [comment]
Try adding `-DUSE_LD_GOLD=OFF` to `TERMUX_PKG_EXTRA_CONFIGURE_ARGS`.

**Comment by noobiescoder** [comment]
> Try adding `-DUSE_LD_GOLD=OFF` to `TERMUX_PKG_EXTRA_CONFIGURE_ARGS`.

I'll try

File: packages/solidity/build.sh [hunk 1]
```diff
@@ -2,11 +2,11 @@ TERMUX_PKG_HOMEPAGE=https://solidity.readthedocs.io
 TERMUX_PKG_DESCRIPTION="An Ethereum smart contract-oriented language."
 TERMUX_PKG_LICENSE="GPL-3.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION=0.8.9
+TERMUX_PKG_VERSION=0.8.11
 TERMUX_PKG_REVISION=1
```
Comment (xtkoba) [inline packages/solidity/build.sh:6]: Please reset (remove) `TERMUX_PKG_REVISION` when bumping `TERMUX_PKG_VERSION`.

**Comment by xtkoba** [comment]
(Also the current `TERMUX_PKG_DEPENDS` seems to be questionable, but it is not relevant to the changes proposed in this PR. We will try to fix this afterwards.)

**Pull request merged**